src/thread/thread_entry_task.hpp

```diff
@@ -133,6 +132,31 @@ namespace cubthread
       }
   };
 
+  // system_worker_entry_manager
+  //
+  // description:
+  //    - generic context manager implementation can be used to provide custom
+  //      thread type and transaction index
+  //    - useful in scenarios where there is no actual transaction and perf logging
+  //      still needs a non-null transaction index
+  //
+  class system_worker_entry_manager : public entry_manager
+  {
+    public:
+      system_worker_entry_manager (thread_type a_thread_type);
+      system_worker_entry_manager (const system_worker_entry_manager &) = delete;
+      system_worker_entry_manager (system_worker_entry_manager &&) = delete;
```

Comment:
Have you considered to make these move-able? Like std::thread?

Reply:
There's no need currently to have them moveable. So, instead they are explicitly restricted.
The usage pattern is that the `cubthread::manager::create_worker_pool` take a raw pointer wo ownership (it does have an internal `m_entry_manager` which it owns) and thus the - currently - single use case does not mandate move semantics.
To make it more streamlined, changes would need to start in `cubthread::manager`.
Did I get your comment straight?

Reply:
Yes, you got it right.